### ISSUE TITLE
Not detecting php syntax errors

### ISSUE THREAD

**Issue by dmxd5** [squizlabs/PHP_CodeSniffer#2930]
I installed the `PHP_CodeSniffer` via `git clone` and set the coding standard to PSR12.

The coding standard error detecting seems to be working fine, but it does not report on syntax errors.

I've tried setting `php_path` in the configuration, but it did not help:
```
phpcs --config-set php_path C:\php
```
I'm on Windows 10 and the path to `php` is also set in Environment Variable.

**Comment by jrfnl** [comment]
@dmxd5 PHPCS is not a linter. It doesn't report on syntax errors _unless you explicitly ask for it_ and even then it uses the PHP `-l` option to do so.

If you really want to use PHPCS to get your syntax errors, add the `Generic.PHP.Syntax` sniff to your custom ruleset.

Or just use `php -l` or a dedicated CLI tool for this, like [PHP Parallel Lint](https://github.com/php-parallel-lint/PHP-Parallel-Lint/).

**Comment by dmxd5** [comment]
> **PHPCS is not a linter.**

I though it uses some built-in linter as described in the [documentation](https://github.com/squizlabs/PHP_CodeSniffer/wiki/Configuration-Options#setting-the-path-to-php):
_The Generic coding standard includes a sniff that will check the syntax of each PHP file using the built-in PHP linter. Use the php_path configuration option to tell the Syntax sniff where to find the PHP binary._

> **add the Generic.PHP.Syntax sniff to your custom ruleset.**

How and where exactly do I add it, providing that I use PSR12 standard?

**Comment by jrfnl** [comment]
> I though it uses some built-in linter as described in the documentation:

Yes, the **PHP native** build in linter.

> How and where exactly do I add it, providing that I use PSR12 standard?

Add a file called `phpcs.xml.dist` to your project root with the following content:
```xml
<?xml version="1.0"?>
<ruleset name="My Project">
    <description>My Project rules for PHP_CodeSniffer</description>
    <rule ref="PSR12"/>
    <rule ref="Generic.PHP.Syntax"/>
</ruleset>
```

PHPCS will pick up on the file automatically, so you don't need to pass the `--standard=` CLI parameter anymore.

**Comment by dmxd5** [comment]
@jrfnl it works! Thank you very much for your assistance. Just one quick question if I may: is there a way to set this linting to phpcs globally by default, and not on per project basis? If so, how do I do that?

**Comment by jrfnl** [comment]
@dmxd5 That would require creating a named ruleset and using that for your projects (or including that in the project ruleset). Explaining that is not something for a "quick question".

To be honest, I honestly think that for linting you'd be better off with PHP Parallel Lint. (see link above)
